### ISSUE TITLE
R1: IDF-weighted FLOPS in _compute_flops_v29()

### ISSUE THREAD

**Issue by mateon01** [sa-project-private-repo/opensearch-neural-pre-train#21]
## Problem

`SPLADELossV29._compute_flops_v29()` uses **unweighted** FLOPS regularization:
```python
mean_activation = sparse_repr.mean(dim=0)
flops_loss = (mean_activation ** 2).sum()  # All tokens penalized equally
```

This treats Korean particles (을, 에, 를, 의) the same as semantic content tokens, resulting in no differential suppression of stopwords.

## Root Cause

V26 `IDFAwareFLOPSLoss` has `penalty_weights` with `stopword_penalty=15x` and `special_penalty=100x`, but V29 bypasses this entirely by using its own unweighted FLOPS computation (lambda_flops=0.0 disables V26 FLOPS).

## Solution

Modify `_compute_flops_v29()` to use IDF-weighted penalty from V26's `self.flops_loss.penalty_weights`:

```python
def _compute_flops_v29(self, sparse_repr):
    mean_activation = sparse_repr.mean(dim=0)
    penalty_weights = self.flops_loss.penalty_weights
    flops_loss = (penalty_weights * (mean_activation ** 2)).sum()
    return flops_loss
```

## Priority
Critical - Primary cause of Korean particle domination in V30 inference results.

## Related
- #19 (top_k_sparse masking bug)
- #20 (V30 dataset + training)
